libbeat/logp/encoding.go

```diff
@@ -0,0 +1,66 @@
+package logp
+
+import (
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+func buildEncoder(cfg Config) zapcore.Encoder {
+	if cfg.JSON {
+		return zapcore.NewJSONEncoder(jsonEncoderConfig())
+	} else if cfg.ToSyslog {
+		return zapcore.NewConsoleEncoder(syslogEncoderConfig())
+	} else {
+		return zapcore.NewConsoleEncoder(consoleEncoderConfig())
+	}
+}
+
+func jsonEncoderConfig() zapcore.EncoderConfig {
+	return zapcore.EncoderConfig{
+		TimeKey:        "timestamp",
+		LevelKey:       "level",
+		NameKey:        "logger",
+		CallerKey:      "caller",
+		MessageKey:     "message",
+		StacktraceKey:  "stacktrace",
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.LowercaseLevelEncoder,
+		EncodeTime:     zapcore.ISO8601TimeEncoder,
+		EncodeDuration: millisecondsDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+		EncodeName:     zapcore.FullNameEncoder,
+	}
+}
+
+func consoleEncoderConfig() zapcore.EncoderConfig {
```

Comment:
I wonder if we should have a base config like the `consoleEncoderConfig` and to the changes on top of it the same way you did for the syslogEncoder but also the jsonEncoder? It seems the jsonEncoder has also only 2 differences? Would make it easier to figure out what we do as a "base level" and which changes are specific to a logger.

Reply:
I agree and noticed this myself that there was some duplication. I can clean this up in another PR.